### DISCUSSION TITLE
Client component code should run on server on dev page load?

### DISCUSSION THREAD

**Discussion by imLymei** [vercel/next.js#69984] [Help] [answered]
### Summary

I have a console.log on my client component but it is logging on the server terminal when my page load on `npm run dev`, is this expected?

![image](https://github.com/user-attachments/assets/d1316cb8-131a-4292-9e47-c8d9e88d1d59)

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Yes, Client Components are also SSR'd. 

- https://github.com/reactwg/server-components/discussions/4

The `use client` directive, doesn't mean that the component is only run client side, rather, it means that the JavaScript for this module is sent to the client.

**Reply by imLymei** [reply]
Thanks for the response! This is actually very interesting because none of my older projects had this. You can find a lot of tutorials on YouTube about Next.js that use examples like this to show that the code only runs on the client. Because of this, I thought it wasn't intentional. Did something recent cause this change? If not, why didn’t my old projects behave this way? Again, thanks for responding to me earlier.

**Reply by icyJoseph** [reply]
No, this has been the behavior since day 1 of React server components (including of course, Client Components).

As a matter of fact, Client Components is what we have had this far, say in Pages Router, or any other React framework. The new thing are Server components that only run server side. 

Why wouldn't you see logs on other projects, I don't know. Probably some config to silence logs, or perhaps you lazy loaded, or deferred your client components.

- https://demystifying-rsc.vercel.app/client-components/next-dynamic/
- https://demystifying-rsc.vercel.app/client-components/no-ssr/

**Reply by imLymei** [reply]
Thank you for the answers! That makes sense. Thinking back, it's just like you said client's components have hydratation errors because they also run on the server. I don’t understand why they unsilenced the logs on the server, though, but everything else makes total sense. Thank you!
